prophet-bounded-context/src/compat.rs

```diff
@@ -0,0 +1,132 @@
+use derive_new::new;
+use prophet_model::{DatabaseType, Entity, Field, Microservice};
+use serde::{Deserialize, Serialize};
+
+/// Request DTO:
+#[derive(new, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct BoundedContextRequest {
+    context: BoundedContextSystem,
+    use_wu_palmer: bool,
+}
+
+#[derive(new, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct BoundedContextSystem {
+    system_name: String,
+    modules: Vec<BoundedContextModule>,
+}
+
+#[derive(new, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct BoundedContextModule {
+    name: String,
+    entities: Vec<BoundedContextEntity>,
+}
+
+#[derive(new, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct BoundedContextEntity {
+    entity_name: String,
+    fields: Vec<BoundedContextField>,
+}
+
+#[derive(new, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct BoundedContextField {
+    name: String,
+    r#type: String,
+}
+
+impl From<Microservice> for BoundedContextSystem {
+    fn from(ms: Microservice) -> Self {
+        BoundedContextSystem {
+            system_name: ms.name,
+            modules: ms
+                .ref_entities
+                .into_iter()
+                .map(|entity| BoundedContextModule::new(entity.name.clone(), vec![entity.into()]))
+                .collect(),
+        }
+    }
+}
+
+impl From<Entity> for BoundedContextEntity {
+    fn from(entity: Entity) -> Self {
+        BoundedContextEntity {
+            entity_name: entity.name,
+            fields: entity
+                .fields
+                .into_iter()
+                .map(|field| field.into())
+                .collect(),
+        }
+    }
+}
+
+impl From<Field> for BoundedContextField {
+    fn from(field: Field) -> Self {
+        BoundedContextField {
+            name: field.name,
+            r#type: field.ty,
+        }
+    }
+}
+
+/// Response DTO:
+#[derive(Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+pub(crate) struct MergedEntitySystem {
+    system_name: String,
```

Comment:
If these fields should be here even though they're not used go ahead and just add `#[allow(unused)]` to them

Reply:
Yeah. They aren't used currently, but are part of the response from the microservice (so it felt weird to ignore them).